### PR TITLE
[Flight] Parse Stack Trace from Structured CallSite if available

### PR DESCRIPTION
**Pull request by sebmarkbage**
This is first step to include more enclosing line/column in the parsed data.

We install our own `prepareStackTrace` to collect structured callsite data and only fall back to parsing the string if it was already evaluated or if `prepareStackTrace` doesn't work in this environment.

We still mirror the default V8 format for encoding the function name part. A lot of this is covered by tests already.

### CODE REVIEW

File: packages/react-server/src/ReactFlightStackConfigV8.js [hunk 1]
```diff
@@ -9,23 +9,104 @@
...
+          filename = origin.toString() + ', <anonymous>';
+        }
+      }
+      const line = callSite.getLineNumber() || 0;
+      const col = callSite.getColumnNumber() || 0;
+      result.push([name, filename, line, col]);
+    }
   }
+  // At the same time we generate a string stack trace just in case someone
+  // else reads it. Ideally, we'd call the previous prepareStackTrace to
```
Comment (eps1lon) [inline packages/react-server/src/ReactFlightStackConfigV8.js:95]: We might have to call the previous one regardless in case the previous one is a custom one that also contains side-effects. Or do that in Next.js depending on the timing. The tricky bit is figuring how if that's the native one we want to avoid because it would be wasteful to call that.

File: packages/react-server/src/ReactFlightStackConfigV8.js [hunk 2]
```diff
@@ -9,23 +9,104 @@
...
-  // eagerly on the server. If the stack has already been read, then we might
-  // not get a normalized stack and it might still have been source mapped.
-  const previousPrepare = Error.prepareStackTrace;
-  Error.prepareStackTrace = DefaultPrepareStackTrace;
-  try {
-    // eslint-disable-next-line react-internal/safe-string-coercion
-    return String(error.stack);
-  } finally {
-    Error.prepareStackTrace = previousPrepare;
+const identifierRegExp = /^[a-zA-Z_$][0-9a-zA-Z_$]*$/;
```
Comment (eps1lon) [inline packages/react-server/src/ReactFlightStackConfigV8.js:15]: Why do we need to test that? Asking because people already complain that our lint rules don't allow unicode characters in Component names so this may come up again here.

**Review by eps1lon — APPROVED** [review]

File: packages/react-server/src/ReactFlightStackConfigV8.js (hunk 2 above)
Comment (sebmarkbage) [reply packages/react-server/src/ReactFlightStackConfigV8.js:15]: I'm trying to mirror what V8 does as much as possible so we get the same from parsing and not.

https://github.com/v8/v8/blob/main/src/objects/call-site-info.cc#L767

However, I didn't dig into what isIdentifier does since you get lost in codegen. I just saw others do this. Usually what I do is this:

https://github.com/facebook/react/blob/main/packages/react-reconciler/src/ReactFiberHydrationDiffs.js#L212-L213
